client/src/lib.rs

```diff
@@ -155,9 +158,17 @@ pub async fn aggregator_hpke_config(
         )));
     }
 
-    Ok(HpkeConfig::decode(&mut Cursor::new(
+    let hpke_configs = HpkeConfigList::decode(&mut Cursor::new(
         hpke_config_response.bytes().await?.as_ref(),
-    ))?)
+    ))?;
+
+    // TODO(#857): Pick one of the advertised HPKE configs. For now, just take the first one, since
+    // we support any HpkeConfig we can decode, and it should be the server's preferred one.
+    let hpke_config = VecDeque::from(hpke_configs.as_vec()).pop_front().ok_or(
```

Comment:
Is the goal to just get the first `hpke_config`? If so, I think you could state this more simply with something like `hpke_configs.hpke_configs().get(0)` (and then `clone` the result, I think).

Reply:
(alternately: `hpke_configs.as_vec().into_iter().next().ok_or(...)?`

Reply:
Yes, I was trying to avoid the `clone()`, but I suspect it's the sort of thing that gets optimized away, and it's not worth carrying around the extra method on `HpkeConfigList`.